plugins/outputs/opensearch/opensearch.go

```diff
@@ -0,0 +1,451 @@
+	EnableGzip          bool            `toml:"enable_gzip"`
+	EnableSniffer       bool            `toml:"enable_sniffer"`
+	FloatHandling       string          `toml:"float_handling"`
+	FloatReplacement    float64         `toml:"float_replacement_value"`
+	ForceDocumentID     bool            `toml:"force_document_id"`
+	HealthCheckInterval config.Duration `toml:"health_check_interval"`
+	HealthCheckTimeout  config.Duration `toml:"health_check_timeout"`
+	IndexName           string          `toml:"index_name"`
+	ManageTemplate      bool            `toml:"manage_template"`
+	OverwriteTemplate   bool            `toml:"overwrite_template"`
+	Password            config.Secret   `toml:"password"`
+	TemplateName        string          `toml:"template_name"`
+	Timeout             config.Duration `toml:"timeout"`
+	URLs                []string        `toml:"urls"`
+	UsePipeline         string          `toml:"use_pipeline"`
+	Username            config.Secret   `toml:"username"`
+	Log                 telegraf.Logger `toml:"-"`
+	pipelineName        string
+	pipelineTagKeys     []string
+	tagKeys             []string
+	onSucc              func(context.Context, opensearchutil.BulkIndexerItem, opensearchutil.BulkIndexerResponseItem)
+	onFail              func(context.Context, opensearchutil.BulkIndexerItem, opensearchutil.BulkIndexerResponseItem, error)
+	httpconfig.HTTPClientConfig
+	osClient *opensearch.Client
+}
+
+//go:embed telegrafTemplate.json
+var telegrafTemplate string
+
+type templatePart struct {
+	TemplatePattern string
+}
+
+func (*Opensearch) SampleConfig() string {
+	return sampleConfig
+}
+
+func (o *Opensearch) Init() error {
+	if o.URLs == nil || o.IndexName == "" {
+		return fmt.Errorf("opensearch urls or index_name is not defined")
+	}
+
+	// Determine if we should process NaN and inf values
+	valOptions := []string{"", "none", "drop", "replace"}
+	if err := choice.Check(o.FloatHandling, valOptions); err != nil {
+		return fmt.Errorf("invalid float_handling type %q", o.FloatHandling)
+	}
+
+	if o.FloatHandling == "" {
+		o.FloatHandling = "none"
+	}
+
+	o.IndexName, o.tagKeys = o.GetReplacementKeys(o.IndexName, ".Tag", "%s")
+	o.pipelineName, o.pipelineTagKeys = o.GetReplacementKeys(o.UsePipeline, "", "%s")
+
+	o.onSucc = func(ctx context.Context, item opensearchutil.BulkIndexerItem, res opensearchutil.BulkIndexerResponseItem) {
+		o.Log.Debugf("Indexed to OpenSearch with status- [%d] Result- %s DocumentID- %s \n", res.Status, res.Result, res.DocumentID)
+	}
+	o.onFail = func(ctx context.Context, item opensearchutil.BulkIndexerItem, res opensearchutil.BulkIndexerResponseItem, err error) {
+		if err != nil {
+			o.Log.Errorf("error while OpenSearch bulkIndexing: %w", err)
+		} else {
+			o.Log.Errorf("error while OpenSearch bulkIndexing: %s: %s", res.Error.Type, res.Error.Reason)
+		}
+	}
+
+	if o.TemplateName == "" {
+		return fmt.Errorf("OpenSearch template_name configuration not defined")
```

Comment:
Error messages should start lowercase in Golang, so either reword or remove the `OpenSearch` prefix, the plugin name is already added by the logger...

Reply:
done